quickwit-config/src/config.rs

```diff
@@ -377,84 +395,65 @@ mod tests {
-            let server_config = serde_yaml::from_str::<ServerConfig>(server_config_yaml).unwrap();
-            assert_eq!(server_config.version, 0);
+            let quickwit_config =
+                serde_yaml::from_str::<QuickwitConfig>(server_config_yaml).unwrap();
+            assert_eq!(quickwit_config.version, 0);
+            assert_eq!(quickwit_config.node_id, "1");
             assert_eq!(
-                server_config.metastore_uri,
+                quickwit_config.metastore_uri,
                 "postgres://username:password@host:port/db"
             );
-            assert!(server_config.indexer_config.is_none());
-            assert!(server_config.searcher_config.is_none());
-            assert!(server_config.storage_config.is_none());
+            assert!(quickwit_config.storage_config.is_none());
         }
         {
             let server_config_yaml = r#"
             version: 0
             metastore_uri: postgres://username:password@host:port/db
- 
-            indexer:
-              data_dir_path: /opt/quickwit/data
-
-            searcher:
-              data_dir_path: /opt/quickwit/data
-              host_key_path: /opt/quickwit/host_key
+            data_dir_path: /opt/quickwit/data
         "#;
-            let server_config = serde_yaml::from_str::<ServerConfig>(server_config_yaml).unwrap();
+            let server_config = serde_yaml::from_str::<QuickwitConfig>(server_config_yaml).unwrap();
             assert_eq!(server_config.version, 0);
             assert_eq!(
                 server_config.metastore_uri,
                 "postgres://username:password@host:port/db"
             );
             assert_eq!(
-                server_config.indexer_config.unwrap(),
+                server_config.indexer_config,
                 IndexerConfig {
-                    data_dir_path: PathBuf::from("/opt/quickwit/data"),
                     ..Default::default()
                 }
             );
             assert_eq!(
-                server_config.searcher_config.unwrap(),
+                server_config.searcher_config,
                 SearcherConfig {
-                    data_dir_path: PathBuf::from("/opt/quickwit/data"),
-                    host_key_path: PathBuf::from("/opt/quickwit/host_key"),
                     ..Default::default()
                 }
             );
         }
-    }
-
-    #[test]
-    fn test_indexer_config_validate() {
         {
-            let indexer_config = IndexerConfig {
-                data_dir_path: "/data/dir/does/not/exist".into(),
-                ..Default::default()
-            };
-            assert!(indexer_config.validate().is_err());
-        }
-        {
-            let indexer_config = IndexerConfig {
-                data_dir_path: env::current_dir().unwrap(),
-                ..Default::default()
-            };
-            assert!(indexer_config.validate().is_ok());
+            let minimal_config_yaml = "version: 0";
+            let server_config =
```

Comment:
`s/server_config/config`?

Reply:
I'm renaming everything, will commit that soon.